### PR TITLE
TINKERPOP-2420 Per Request Options in .NET

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-2420

Added per request options in .NET. Opted to not create `RequestOptions` like functionality in .NET since the way the client is exposed allows you to construct a bare `RequestMessage` with reasonable niceness - documented that approach a bit better. Also added documentation for the old GLVs that recently got this feature. 

VOTE +1

### CODE REVIEW

File: gremlin-dotnet/src/Gremlin.Net/Driver/Tokens.cs [hunk 1]
```diff
@@ -74,24 +79,25 @@ public class Tokens
         public static string ProcessorSession = "session";
 
         /// <summary>
-        ///     Argument name that allows to defines the number of iterations each ResponseMessage should contain - overrides the
-        ///     resultIterationBatchSize server setting.
+        ///     Argument name that allows to definition of the number of iterations each ResponseMessage should
```
Comment (FlorianHockmann) [inline gremlin-dotnet/src/Gremlin.Net/Driver/Tokens.cs:82]: (nitpick) _allows to definition_ doesn't sound right to me (?)

File: gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Process/Traversal/DriverRemoteConnection/GraphTraversalTests.cs [hunk 2]
```diff
@@ -201,22 +204,21 @@ public void ShouldUseBindingsInTraversal()
...
                 {"y", true}
             };
             var g = AnonymousTraversalSource.Traversal().WithRemote(connection);
-
-            var b = new Bindings();
+            
             var countWithStrategy = g.WithStrategies(new OptionsStrategy(options)).V().Count().Next();
             Assert.Equal(6, countWithStrategy);
+            
+            Assert.Equal(ResponseStatusCode.ServerTimeout, Assert.Throws<ResponseException>(() =>
```
Comment (FlorianHockmann) [inline gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Process/Traversal/DriverRemoteConnection/GraphTraversalTests.cs:218]: (nitpick) I think it's easier to read and understand this when the two asserts are separated:

```c#
var responseException = Assert.Throws<ResponseException>(() =>
    g.With("y").With("x", "test").With(Tokens.ArgsEvalTimeout, 10).Inject(1)
        .SideEffect(Lambda.Groovy("Thread.sleep(10000)")).Iterate());
Assert.Equal(ResponseStatusCode.ServerTimeout, responseException.StatusCode);
```

File: docs/src/reference/gremlin-variants.asciidoc [hunk 3]
```diff
@@ -1121,6 +1121,18 @@ the remote end.
 include::../../../gremlin-dotnet/test/Gremlin.Net.IntegrationTest/Docs/Reference/GremlinVariantsTests.cs[tags=connecting]
 ----
 
+Some connection options can also be set on individual requests made through the using `With()` step on the
```
Comment (FlorianHockmann) [inline docs/src/reference/gremlin-variants.asciidoc:1124]: _the using `With()` step_ -> _using the `With()` step_

File: docs/src/reference/gremlin-variants.asciidoc [hunk 4]
```diff
@@ -1121,6 +1121,18 @@ the remote end.
...
+`TraversalSource`. For instance to set request timeout to 500 milliseconds:
+
+[source,csharp]
+----
+var l = g.With(Tokens.ArgsEvalTimeout, 500).V().Out("knows").Count();
```
Comment (FlorianHockmann) [inline docs/src/reference/gremlin-variants.asciidoc:1129]: I'm not sure about this but shouldn't we use a terminal step here to actually get the count(s)? Especially new users would probably expect here to have the count stored in the variable and not a traversal object.

File: docs/src/reference/gremlin-variants.asciidoc [hunk 5]
```diff
@@ -1246,6 +1258,20 @@ var gremlinServer = new GremlinServer("localhost", 8182);
 var client = new GremlinClient(gremlinServer, sessionId: Guid.NewGuid().ToString()))
 ----
 
+==== Per Request Settings
+
+The `GremlinClient.submit()` functions accept an option to build a raw `RequestMessage`. A good use-case for this
```
Comment (FlorianHockmann) [inline docs/src/reference/gremlin-variants.asciidoc:1263]: (nitpick) **S**ubmit (to stay with the .NET naming conventions)

**Review by FlorianHockmann — COMMENTED** [review]
LGTM, I only found a few nits and I also agree that `RequestOptions` shouldn't be necessary here, especially since we want to discourage people from using Gremlin scripts in the first place so that doesn't need to be so easy to use like the traversal API.

**Comment by spmallette** [comment]
Addressed all your comments @FlorianHockmann - thanks!

**Review by FlorianHockmann — APPROVED** [review]
VOTE +1

**Pull request merged**